feature-set/Cargo.toml

```diff
@@ -19,8 +18,14 @@ solana-frozen-abi = { workspace = true, optional = true, features = [
 solana-frozen-abi-macro = { workspace = true, optional = true, features = [
     "frozen-abi",
 ] }
-solana-hash = { workspace = true }
 solana-pubkey = { workspace = true }
+
+[target.'cfg(target_os = "solana")'.dependencies]
+ahash = { workspace = true, default-features = false, features = ["std", "no-rng"] }
+
+[target.'cfg(not(target_os = "solana"))'.dependencies]
+ahash = { workspace = true, default-features = true }
+solana-hash = { workspace = true }
 solana-sha256-hasher = { workspace = true }
```

Comment:
can you explain why solana-hash and sha256-hasher are moved under the not(solana) scope?

Reply:
nvm i see now below - the generated ID only needed for validator.